packages/harmony/src/components/layout/Popup/Popup.tsx

```diff
@@ -0,0 +1,411 @@
+      const left = anchorRect.x + anchorTranslation.x - wrapperTranslation.x
+
+      const { adjustedTop, adjustedLeft } = getAdjustedPosition(
+        top,
+        left,
+        wrapperRect
+      )
+
+      if (wrapperRef.current) {
+        wrapperRef.current.style.top = `${top + adjustedTop}px`
+        wrapperRef.current.style.left = `${left + adjustedLeft}px`
+      }
+
+      originalTopPosition.current = top
+    }
+  }, [
+    isVisible,
+    wrapperRef,
+    anchorRef,
+    anchorOrigin,
+    transformOrigin,
+    setComputedTransformOrigin,
+    originalTopPosition,
+    containerRef
+  ])
+
+  // Callback invoked on each scroll. Uses original top position to scroll with content.
+  // Takes scrollParent to get the current scroll position as well as the intitial scroll position
+  // when the popup became visible.
+  const watchScroll = useCallback(
+    (scrollParent: Element, initialScrollPosition: number) => {
+      const scrollTop = scrollParent.scrollTop
+      if (wrapperRef.current) {
+        wrapperRef.current.style.top = `${
+          originalTopPosition.current - scrollTop + initialScrollPosition
+        }px`
+      }
+    },
+    [wrapperRef, originalTopPosition]
+  )
+
+  // Set up scroll listeners
+  useEffect(() => {
+    if (isVisible && anchorRef.current) {
+      const scrollParent = getScrollParent(anchorRef.current)
+      if (!scrollParent) return
+
+      const initialScrollPosition = scrollParent.scrollTop
+      const listener = () => watchScroll(scrollParent, initialScrollPosition)
+      scrollParent.addEventListener('scroll', listener)
+      return () => {
+        scrollParent.removeEventListener('scroll', listener)
+      }
+    }
+
+    return () => {}
+  }, [isVisible, watchScroll, anchorRef])
+
+  // Set up key listeners
+  useEffect(() => {
+    if (isVisible) {
+      const escapeListener = (e: KeyboardEvent) => {
+        if (e.code === 'Escape') {
+          handleClose()
+        }
+      }
+
+      window.addEventListener('keydown', escapeListener)
+
+      return () => window.removeEventListener('keydown', escapeListener)
+    }
+    return () => {}
+  }, [isVisible, handleClose])
+
+  const transitions = useTransition(isVisible, null, {
+    from: {
+      transform: `scale(0)`,
+      opacity: 0
+    },
+    enter: {
+      transform: `scale(1)`,
+      opacity: 1
+    },
+    leave: {
+      transform: `scale(0)`,
+      opacity: 0
+    },
+    config: theme.spring.standard,
+    unique: true
+  })
+
+  const rootStyle = zIndex ? { zIndex } : {}
+
+  const handleMouseLeave = useCallback(() => {
+    if (dismissOnMouseLeave) {
+      onClose?.()
+    }
+  }, [dismissOnMouseLeave, onClose])
+
+  return (
+    <>
+      {/* Portal the popup out of the dom structure so that it has a separate stacking context */}
+      {ReactDOM.createPortal(
+        <div
+          ref={wrapperRef}
+          className={styles.root}
+          style={rootStyle}
```

Comment:
can use `css` prop for this ([example](https://github.com/AudiusProject/audius-protocol/blob/main/packages/harmony/src/components/button/Button/Button.stories.tsx#L65-L66)) but this is prob fine